### PR TITLE
feat(Dialog): add prop contentOverflow to control overflow behaviour

### PR DESCRIPTION
**Pull request by DenisVershkov**
1. Added prop 'contentOverflow' to the Modal and Dialog. The prop handles scroll appearance:

- if 'auto' is selected, then scroll will be applied to the DialogBody
- if 'visible' is selected, then scroll will appear outside (the same behaviour as before)
- 'visible' is the default value

2. Added prop 'hasDivider' to the DialogBody component. When this prop value is 'true', then DialogBody has top and bottom border. I've done this for the contentOverflow 'auto' case, so it should be used like: <DialogBody hasDivider={contentOverflow === 'auto'}>

### CODE REVIEW

File: src/components/Dialog/Dialog.scss [hunk 1]
```diff
@@ -4,6 +4,15 @@ $block: '.#{variables.$ns}dialog';
 
 #{$block} {
     position: relative;
+    display: flex;
+    flex-direction: column;
+
+    &_has-scroll {
+        overflow-y: auto;
+        max-height: calc(
+            100vh - calc(var(--g-modal-margin, #{variables.$modal-default-margin})) * 2
```
Comment (amje) [inline src/components/Dialog/Dialog.scss:13]: 2nd calc is just wraps var?

File: src/components/Dialog/DialogBody/DialogBody.tsx [hunk 2]
```diff
@@ -9,10 +9,11 @@ const b = block('dialog-body');
 export interface DialogBodyProps {
     children: React.ReactNode;
     className?: string;
+    hasDividers?: boolean;
```
Comment (amje) [inline src/components/Dialog/DialogBody/DialogBody.tsx:12]: Wouldn't be `hasBorders` more precise? Divider - smth that splits the content, border - is on the edge. While it's a `DialogBody` prop, borders suits better imo.

File: src/components/Modal/Modal.scss [hunk 3]
```diff
@@ -16,24 +16,29 @@ $block: '.#{variables.$ns}modal';
...
     &__content {
-        display: inline-block;
-        margin: var(--g-modal-margin, 20px);
-        text-align: start;
         background-color: var(--g-color-base-modal);
-        border-radius: var(--g-modal-border-radius, 5px);
+        &_has-scroll {
+            overflow-y: auto;
+            max-height: calc(
+                100vh - calc(var(--g-modal-margin, #{variables.$modal-default-margin})) * 2
```
Comment (amje) [inline src/components/Modal/Modal.scss:39]: Same comment as above

File: src/components/variables.scss [hunk 4]
```diff
@@ -5,3 +5,5 @@ $s-height: 24px;
 $m-height: 28px;
 $l-height: 36px;
 $xl-height: 44px;
+$modal-default-border-radius: 5px;
```
Comment (amje) [inline src/components/variables.scss:8]: While we're only using it inside the Modal, we don't need to share it.

**Review by amje — CHANGES_REQUESTED** [review]

**Review by amje — APPROVED** [review]

**Comment by DenisVershkov** [comment]
closes [DATAUI-1254]

**Pull request merged**